src/lib.rs

```diff
@@ -90,17 +91,25 @@ impl<F: PrimeField, PC: PolynomialCommitment<F>, D: Digest> Marlin<F, PC, D> {
         srs
     }
 
-    /// Generate the index-specific (i.e., circuit-specific) prover and verifier
-    /// keys. This is a deterministic algorithm that anyone can rerun.
-    pub fn index<C: ConstraintSynthesizer<F>>(
-        committer_key: &PC::CommitterKey,
-        c: C,
-    ) -> Result<(ProverKey<F, PC>, VerifierKey<F, PC>), Error<PC::Error>> {
+    /// Wrapper around `AHPForR1CS::index`, for external access
+    pub fn get_index_info<C: ConstraintSynthesizer<F>>(c: C) -> Result<Index<F>, Error<PC::Error>> {
```

Comment:
A bit misleading name: ``get_index_info`` does re-run the indexer, hence it is not merely a lookup.

Reply:
You're right but I don't want to confuse it with 'index'. Actually index function should be renamed according to me but I don't want to do it now cause I'll need to refactor in too many places